presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/TemporalFunctionFactory.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.storage.organization;
+
+import com.facebook.presto.raptor.storage.StorageManagerConfig;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.SqlDate;
+import com.facebook.presto.spi.type.SqlTimestamp;
+import com.facebook.presto.spi.type.Type;
+import org.joda.time.DateTimeZone;
+
+import javax.inject.Inject;
+
+import java.time.Duration;
+
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class TemporalFunctionFactory
+{
+    private final DateTimeZone shardDayBoundaryTimeZone;
+
+    @Inject
+    public TemporalFunctionFactory(StorageManagerConfig config)
+    {
+        this.shardDayBoundaryTimeZone = requireNonNull(config.getShardDayBoundaryTimeZone(), "shardDayBoundaryTimeZone is null");
+    }
+
+    public TemporalFunction createTemporalFunction(Type type)
+    {
+        if (type == DATE) {
+            return new DateTemporalFunction();
```

Comment:
The classes you create are stateless, so you don't need to create a new one on every method call.